looks like arch doesn't like some of the compiler flags (architecture extensions), if arch is not a supported architecture, then I would change meta.platforms to reflect just the x86_64 variants

```
/nix/store/nalgbdpjmhizap3sp1q3lqinisifldl9-gcc-wrapper-8.3.0/bin/cc -Inrepel@sha -I. -I.. -I/nix/store/mwjc8kdg97nbl07bwsr96arkqmiz2v1w-fftw-single-3.3.8-dev/include -I/nix/store/6n1ib7m8axh6k75n6v9nkhbxp7s89ym6-lv2-1.16.0/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=c99 -fPIC -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -fno-finite-math-only -MD -MQ 'nrepel@sha/src_nrepel.c.o' -MF 'nrepel@sha/src_nrepel.c.o.d' -o 'nrepel@sha/src_nrepel.c.o' -c ../src/nrepel.c
gcc: error: unrecognized command line option '-msse'; did you mean '-fdse'?
gcc: error: unrecognized command line option '-msse2'
gcc: error: unrecognized command line option '-mfpmath=sse'
ninja: build stopped: subcommand failed.
builder for '/nix/store/7zy3w7g3mpk39zyr1ag7w8iwg364clg2-noise-repellent-unstable-2018-12-29.drv' failed with exit code 1
error: build of '/nix/store/7zy3w7g3mpk39zyr1ag7w8iwg364clg2-noise-repellent-unstable-2018-12-29.drv' failed
```

--- a/pkgs/applications/audio/noise-repellent/default.nix
+++ b/pkgs/applications/audio/noise-repellent/default.nix
@@ -1,0 +1,29 @@
+{ stdenv, fetchFromGitHub, meson, ninja, pkgconfig, fftwFloat, lv2 }:
+
+stdenv.mkDerivation rec {
+  pname = "noise-repellent";
+  version = "unstable-2018-12-29";
+
+  src = fetchFromGitHub {
+    owner = "lucianodato";
+    repo = pname;
+    rev = "9efdd0b41ec184a792087c87cbf5382f455e33ec";
+    sha256 = "0pn9cxapfvb5l62q86bchyfll1290vi0rhrzarb1jpc4ix7kz53c";
+    fetchSubmodules = true;
+  };
+
+  mesonFlags = ("--prefix=${placeholder "out"}/lib/lv2");
+
+  nativeBuildInputs = [ meson ninja pkgconfig ];
+  buildInputs = [
+    fftwFloat lv2
+  ];
+
+  meta = with stdenv.lib; {
+    description = "An lv2 plugin for broadband noise reduction";
+    homepage    = https://github.com/lucianodato/noise-repellent;
+    license     = licenses.gpl3;
+    maintainers = [ maintainers.magnetophon ];
+    platforms = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "i686-darwin"  ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5248,6 +5248,9 @@ in
 
   nnn = callPackage ../applications/misc/nnn { };
 
+
+  noise-repellent = callPackage ../applications/audio/noise-repellent { };
+
   notary = callPackage ../tools/security/notary { };
 
   notify-osd = callPackage ../applications/misc/notify-osd { };

